### ISSUE TITLE
"panel serve --basic-auth" do not work together with "--prefix"

### ISSUE THREAD

**Issue by tlvu** [holoviz/panel#5487]
`panel serve --basic-auth` do not work together with `--prefix` option.

Basic Auth hardcode the login url to `/login` which do not honor the prefix config.

https://github.com/holoviz/panel/blob/36a1d92921a86803a6028ff7577515f6ba5d83f1/panel/auth.py#L887-L899

Then the `POST` in the login form also hardcode `/login`, which also do not honor the prefix config.

https://github.com/holoviz/panel/blob/36a1d92921a86803a6028ff7577515f6ba5d83f1/panel/_templates/basic_login.html#L80

#### ALL software version info
(this library, plus any other relevant software, e.g. bokeh, python, notebook, OS, browser, etc)

Conda packages

  - bokeh=3.2.2=pyhd8ed1ab_0
  - panel=1.2.1=pyhd8ed1ab_0
  - python=3.10.12=hd12c33a_0_cpython

OS: Linux
Browser: Firefox

#### Description of expected behavior and the observed behavior

Here the redirect should be have been `Location: /prefix/login?next=%2Fprefix`
```
$ curl --silent --include https://HOST/prefix
HTTP/1.1 302 Found
Server: nginx/1.23.4
Date: Fri, 08 Sep 2023 02:08:42 GMT
Content-Type: text/html; charset=UTF-8
Content-Length: 0
Connection: keep-alive
Location: /login?next=%2Fprefix
```

Here it should have been `POST /prefix/login HTTP/1.1`
```
POST /login HTTP/1.1
Host: HOST
User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/117.0
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8
Accept-Language: en-CA,en-US;q=0.8,en;q=0.5,fr-CA;q=0.3
Accept-Encoding: gzip, deflate, br
Content-Type: application/x-www-form-urlencoded
Content-Length: 28
Origin: https://HOST
Connection: keep-alive
Referer: https://HOST/prefix/login
Upgrade-Insecure-Requests: 1
Sec-Fetch-Dest: document
Sec-Fetch-Mode: navigate
Sec-Fetch-Site: same-origin
Sec-Fetch-User: ?1
```

#### Complete, minimal, self-contained example code that reproduces the issue

```
panel serve --basic-auth my_password --cookie-secret secret --allow-websocket-origin=* --prefix=/prefix app.py

```

**Comment by kuykendall-ben** [comment]
I believe this is a dup of https://github.com/holoviz/panel/issues/5398?

**Comment by tlvu** [comment]
> I believe this is a dup of #5398?

Probably.  It means it has not been fixed completely since it's still not working properly in the current latest panel-1.2.2.

**Comment by tlvu** [comment]
> > I believe this is a dup of #5398?
> 
> Probably. It means it has not been fixed completely since it's still not working properly in the current latest panel-1.2.2.

After further testing, I found out the root cause is the relative redirect in the `Location` http response header which seems to only affect some browsers, like Firefox.

It seems initially only absolute redirect are allowed.  But then relative redirects are also allowed but looks like some browsers are not implementing relative redirects properly, see https://stackoverflow.com/questions/8250259/is-a-302-redirect-to-relative-url-valid-or-invalid

So I think to maximize compatibility, absolute redirects should be used.  I've updated the description accordingly.

**Comment by tlvu** [comment]
Further testing again and even Firefox works fine.  It's just the relative redirect that is error-prone.

So if I hit the panel app with `https://HOST/prefix`, then I receive the http response header `Location: ./login?next=/prefix` and the browser subsequently try the wrong redirect `https://HOST/login?next=/prefix`.

However if I hit the panel app with `https://HOST/prefix/` (note the ending slash), then with the same http response header `Location: ./login?next=/prefix`, the browser will subsequently follow the proper redirect  `https://HOST/prefix/login?next=/prefix`. 

In conclusion, I think using absolute redirect (`Location: /prefix/login?next=/prefix` in this example) would be much less error-prone for the users.

**Comment by kuykendall-ben** [comment]
Thanks for investigating further @tlvu! I confirmed, even with other auth methods, that navigating to `https://HOST/prefix/` works correctly and `https://HOST/prefix` does not. I agree with your assessment that absolute redirects would be more flexible. They'll also work correctly if we ever end up with nested directories.
